cli/flox-rust-sdk/src/models/floxmetav2/mod.rs

```diff
@@ -162,15 +183,30 @@ pub fn floxmeta_git_options(floxhub_host: &str, floxhub_token: Option<&str>) ->
     options.add_env_var("GIT_CONFIG_GLOBAL", "/dev/null");
     options.add_env_var("GIT_CONFIG_SYSTEM", "/dev/null");
 
-    if let Some(token) = floxhub_token {
-        // Set authentication with the floxhub token using an inline credential helper.
-        // The credential helper should help avoinding a leak of the token in the process list.
-        options.add_env_var("FLOX_FLOXHUB_TOKEN", token);
-        options.add_config_flag(
-            &format!("credential.{floxhub_host}.helper"),
-            r#"!f(){ echo "username=oauth"; echo "password=$FLOX_FLOXHUB_TOKEN"; }; f"#,
-        );
-    }
+    // set the floxhub origin
+    options.add_config_flag(
+        "remote.dynamicorigin.url",
```

Comment:
question blocking: so is the idea here to always override the remote url, effectively ignoring whatever the remote is? Can you add some comments? It looks like it works for push but maybe not fetch? I don't think calling it `dynamicorigin` is helping me understand what's going on - does that have any specific meaning?
```
~/flox/main main > git -c remote.origin.url="foo" remote -v
origin	git@github.com:flox/flox (fetch)
origin	git@github.com:flox/flox (push)
origin	foo (push)
~/flox/main main > git -c remote.origin.url="foo" push origin main
Everything up-to-date
fatal: 'foo' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
~/flox/main main > git -c remote.origin.url="foo" fetch origin main
From github.com:flox/flox
 * branch              main       -> FETCH_HEAD
```

Reply:
The 'dynamicorigin' remote is literally a dynamic remote that isn't written to the repository.
It is set to either the git url found in the pointer or configured in the config (depending on the operation)
Since the origin can be different for _every_ environment we can't really store it in the repo itself.

We could possibly create git remotes for every environment, but pulling would still have you guess the right host.

Currently we also can't have two environments with the same name from two "floxhubs" I think ( it will work but weirdly overlap)

Another possibility are separate clones for each environment... once again

Reply:
Least I can do is add a comment on what that dymanicorigin thing is, sure

Reply:
Why not have a different repo for each hub? (not a change we need to make right now though)

Reply:
There's not much difference to different repos per hub or throwing it all in one.
The only thing that we rely on is the project branch, i.e. `<name>.<system>.<hash>`

sure with multiple hubs `<name>.<system>.<hash 1>` and `<name>.<system>.<hash 2>` might come from different hubs, but all information regarding where they come from is necessarily required in the pointer (not as configuration, but as metadata).

If we do multiple repos, I'd go one repo per environment as in `floxmeta/<hub-host>/<name>/<system>/<hash>/{repo, link}`...

Reply:
Is this working as expected for fetch? Not sure if the snippet above is exactly what's happening or not?

Reply:
seems so: 

```
↕ 3 /tmp/nix-shell.kzMEvg/tmp.TmlHpnpN5T via ❄️  impure (flox-dev-env) 
❯ git init         
Initialized empty Git repository in /private/tmp/nix-shell.kzMEvg/tmp.TmlHpnpN5T/.git/
↕ 3 tmp.TmlHpnpN5T on  master +/-via ❄️  impure (flox-dev-env) 
❯ git -c "remote.dynamic.url=https://github.com/flox/flox" fetch dynamic      
remote: Enumerating objects: 9488, done.
remote: Counting objects: 100% (898/898), done.
remote: Compressing objects: 100% (463/463), done.
remote: Total 9488 (delta 616), reused 568 (delta 418), pack-reused 8590
Receiving objects: 100% (9488/9488), 5.28 MiB | 26.76 MiB/s, done.
Resolving deltas: 100% (5893/5893), done.
From https://github.com/flox/flox
 * branch            HEAD       -> FETCH_HEAD
❯ git remote -v
↕ 3 tmp.TmlHpnpN5T on  master +/-via ❄️  impure (flox-dev-env) 
❯ 
```

the remote defined in the repository (e.g. origin when cloning) will take precedence but since we use `dynamicorigin` which does not exist in the repo, any operation on remote has access to that.
Note that it also does not save the remote ^

Reply:
Okay perfect wasn't sure if there were overriding concerns